### ISSUE TITLE
New command autosquash?

### ISSUE THREAD

**Issue by lreficent** [OCA/oca-github-bot#86]
Following the discussion here: https://github.com/OCA/oca-github-bot/pull/75, and specifically this clarification: https://github.com/OCA/oca-github-bot/pull/75#issuecomment-545024910.

Personally, when I first red that the bot was going to autosquash commits I loved the feature as I prefer to add fixup commits to have a clear diffs as you attends review or do changes. For me, rely on the force push diff is not ideal. Then today I found that it was not actually working and the merge strategy is a different one, so this is issue raises the question: **is it worth/useful** to implement such an option? something like `ocabot merge autosquash`.

**Comment by pedrobaeza** [comment]
@lreficent what is the problem querying "Forced push" link in GitHub for reviewing differences?

**Comment by lreficent** [comment]
@pedrobaeza if you force push several times in a row you lose the diff. I know it is nitpicking that's why I wonder if the feature is really valuable.

**Comment by pedrobaeza** [comment]
Yeah, that's the only limitation for now, that GitHub groups them together and don't show full diff, but I think we must report that to GitHub for being fixed instead of using other techniques.

**Comment by hparfr** [comment]
Hi 2019! 

End of 2021 here and GitHub has improved a little on forced branch but not so much. After a force push, links from (email) notifications are still broken and it's still difficult to figure out what where changed during second review.

Git already have all the tools to fix it from a while.

From a user point of view. `git commit --fixup HEAD` is easier than `git rebase 14.0`
A short help message can be added to the output of pre-commit on the oca CI:
"Run pre-commit on your machine then commit with `git commit --fixup HEAD` "

In order to ease the transition, I suggest, to allow it as an opt-in :
`ocabot merge minor autosquash`
And if there is some conflicts, it can still be fixed the old way: rebase locally.
